pkg/operator/ceph/controller/spec.go

```diff
@@ -456,3 +463,52 @@ func StoredLogAndCrashVolumeMount(varLogCephDir, varLibCephCrashDir string) []v1
 		},
 	}
 }
+
+// GenerateLivenessProbeExecDaemon makes sure a daemon has a socket and that it can be called and returns 0
+func GenerateLivenessProbeExecDaemon(daemonType, daemonID string) *v1.Probe {
+	confDaemon := getDaemonConfig(daemonType, daemonID)
+
+	return &v1.Probe{
+		Handler: v1.Handler{
+			Exec: &v1.ExecAction{
+				// Run with env -i to clean env variables in the exec context
+				// This avoids conflict with the CEPH_ARGS env
+				//
+				// Example:
+				// env -i sh -c ceph --admin-daemon /run/ceph/ceph-osd.0.asok status
+				Command: []string{
+					"env",
+					"-i",
+					"sh",
+					"-c",
+					fmt.Sprintf("ceph --admin-daemon %s %s", confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand()),
+				},
+			},
+		},
+		InitialDelaySeconds: 10,
+	}
+}
+
+func getDaemonConfig(daemonType, daemonID string) *daemonConfig {
+	return &daemonConfig{
+		daemonType: string(daemonType),
+		daemonID:   daemonID,
+	}
+}
+
+func (c *daemonConfig) buildSocketName() string {
```

Comment:
These are simple enough that why not just pass two string params to the functions? Seems like creating the `daemonConfig` struct and having methods is overkill.

Reply:
Originally, I went with args but felt it was nicer to have the struct so I could avoid passing down the same args multiple times. When I have to pass the same args multiple functions called in the same tree then I feel like the struct is cleaner.

Reply:
Sure this works